app/models/user.rb

```diff
@@ -15,6 +15,16 @@ class User < ActiveRecord::Base
   validates :email, presence: true, uniqueness: { case_sensitive: false }
   validates :password, presence: true, length: { minimum: 6 }, on: :create
 
+  def render_token!
+    self.token = SecureRandom.urlsafe_base64.downcase
+    save
+  end
+
+  def clear_token!
+    self.token = nil
+    save
```

Comment:
If you aim to save immediately, then its better to call `update_column` or `update_attribute` straight.